.github/workflows/java-ec2-default-test.yml

```diff
@@ -93,6 +98,10 @@ jobs:
           if [ "${{ github.event.repository.name }}" = "aws-otel-java-instrumentation" ]; then
             # Reusing the adot-main-build-staging-jar bucket to store the python wheel file
             echo GET_ADOT_JAR_COMMAND="aws s3 cp s3://adot-main-build-staging-jar/aws-opentelemetry-agent.jar ./adot.jar" >> $GITHUB_ENV
+          elif [ "${{ env.OTEL_SOURCE }}" == "maven" ]; then
+            latest_version=$(curl -s "https://search.maven.org/solrsearch/select?q=g:software.amazon.opentelemetry+a:aws-opentelemetry-agent&core=gav&rows=1&wt=json" | jq -r '.response.docs[0].v')
+            echo "Latest version for Maven is: $latest_version"
+            echo GET_ADOT_JAR_COMMAND="wget -O adot.jar https://repo1.maven.org/maven2/software/amazon/opentelemetry/aws-opentelemetry-agent/$latest_version/aws-opentelemetry-agent-$latest_version.jar" >> $GITHUB_ENV
```

Comment:
This should effectively be the same (and a lot harder to implement without breaking existing tests), but shouldn't we do it the way customers would by updating our dependencies? [link](https://mvnrepository.com/artifact/software.amazon.opentelemetry/aws-opentelemetry-agent/1.32.3)
```
// https://mvnrepository.com/artifact/software.amazon.opentelemetry/aws-opentelemetry-agent
implementation group: 'software.amazon.opentelemetry', name: 'aws-opentelemetry-agent', version: '1.32.3'
```